tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/JSDesugaring.scala

```diff
@@ -2245,14 +2247,15 @@ private[emitter] object JSDesugaring {
       implicit outputMode: OutputMode, pos: Position): js.Tree =
     envField("c", className)
 
-  private[emitter] def genRawJSClassConstructor(linkedClass: LinkedClass)(
+  private[emitter] def genRawJSClassConstructor(
+      classEmitter: ScalaJSClassEmitter, linkedClass: LinkedClass)(
       implicit outputMode: OutputMode, pos: Position): js.Tree = {
     if (linkedClass.kind == ClassKind.JSClass) {
       encodeClassVar(linkedClass.encodedName)
     } else {
       require(linkedClass.jsName.isDefined)
       linkedClass.jsName.get.split("\\.").foldLeft(envField("g")) {
-        (prev, part) => js.BracketSelect(prev, js.StringLiteral(part))
+        (prev, part) => classEmitter.genBracketSelect(prev, js.StringLiteral(part))
       }
     }
   }
```

Comment:
Move this into an instance of `JSDesugaring`. Refactor call sites as necessary (or create a forwarder if really necessary).